### ISSUE TITLE
Flush buffer logic causes buffers to be flushed prematurely

### ISSUE THREAD

**Issue by the4thamigo-uk** [airbytehq/airbyte#32346]
### What method are you using to run Airbyte?

Kubernetes

### Platform Version or Helm Chart Version

0.50.10

### What step the error happened?

During the Sync

### Revelant information

Hi, we are using the google sheets source connector together with the s3 sink connector and have noticed some strange behaviour that is causing us some issues. We have a spreadsheet with a number of tabs. Most of them have a handful of records but two have about 300 records. What see is that data from some of the tabs is split into 2 separate files with suffixes _0.parquet and _1.parquet. As mentioned, the volume of data is not huge, and we even have deleted the blank rows on the bottom of each tab, but the data is still being split.

Ive been looking at the code and we are hitting this logic :

https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/core[…]ions/destination/record_buffer/SerializedBufferingStrategy.java](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination/record_buffer/SerializedBufferingStrategy.java#L76-L79

And this is associated with this log entry :

```
2023-11-09 12:42:16 destination > INFO i.a.i.d.r.SerializedBufferingStrategy(flushAllBuffers):133 Flushing all 10 current buffers (160 KB in total)
```

Since the 160kb is well within the limit of the first condition in the `if` block, it looks like we are hitting the second condition : 
https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/core[…]ions/destination/record_buffer/SerializedBufferingStrategy.java](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination/record_buffer/SerializedBufferingStrategy.java#L77C9-L77C10

In other words because we have 10 tabs in our google sheet, as soon as we receive a single record from a 10th concurrent stream, it will flush all the streams (even though we have a tiny amount of data).
What is the rationale for this logic, and is there a workaround, or potential bug fix?

### Relevant log output

```shell
2023-11-09 12:42:16 destination > INFO i.a.i.d.r.SerializedBufferingStrategy(flushAllBuffers):133 Flushing all 10 current buffers (160 KB in total)
```
```
